RecoLocalMuon/CSCSegment/src/CSCSegAlgoRU.h

```diff
@@ -0,0 +1,160 @@
+#ifndef CSCSegment_CSCSegAlgoRU_h
+#define CSCSegment_CSCSegAlgoRU_h
+
+/**
+ * \class CSCSegAlgoRU
+ *
+ * This is the original algorithm for building endcap muon track segments
+ * out of the rechit's in a CSCChamber
+ * 'RU' = 'RUssia' = Road Usage
+ *
+ * A CSCSegment is a RecSegment4D, and is built from
+ * CSCRecHit2D objects, each of which is a RecHit2DLocalPos. <BR>
+ *
+ * This class is used by the CSCSegmentAlgorithm. <BR>
+ * Alternative algorithms can be used for the segment building
+ * by writing classes like this, and then selecting which one is actually
+ * used via the CSCSegmentBuilder. <BR>
+ *
+ * developed and implemented by Vladimir Palichik <Vladimir.Paltchik@cern.ch>
+ *                          and Nikolay Voytishin <nikolay.voytishin@cern.ch>
+ */
+
+#include <RecoLocalMuon/CSCSegment/src/CSCSegmentAlgorithm.h>
+#include <DataFormats/CSCRecHit/interface/CSCRecHit2D.h>
+
+#include <deque>
+#include <Math/Functions.h>
+#include <Math/SVector.h>
+#include <Math/SMatrix.h>
+
+#include <vector>
+
+
+class CSCSegFit;
+
+class CSCSegAlgoRU : public CSCSegmentAlgorithm {
+
+public:
+
+    // Tim tried using map as basic container of all (space-point) RecHit's in a chamber:
+    // The 'key' is a pseudo-layer number (1-6 but with 1 always closest to IP).
+    // The 'value' is a vector of the RecHit's on that layer.
+    // Using the layer number like this removes the need to sort in global z.
+    // Instead we just have to ensure the layer index is correctly adjusted 
+    // to enforce the requirement that 'layer 1' is closest in the chamber
+    // to the IP.
+    
+    /// Typedefs
+  
+  
+    // 4-dim vector                                                                                                                                                                   
+    typedef ROOT::Math::SVector<double,6> SVector6;
+
+    typedef std::vector<int> LayerIndex;
+    typedef std::vector<const CSCRecHit2D*> ChamberHitContainer;
+    typedef std::vector<const CSCRecHit2D*>::const_iterator ChamberHitContainerCIt;
+
+    // We need to be able to flag a hit as 'used' and so need a container
+    // of bool's. Naively, this would be vector<bool>... but AVOID that since it's
+    // non-standard i.e. packed-bit implementation which is not a standard STL container. 
+    // We don't need what it offers and it could lead to unexpected trouble in the future.
+
+    typedef std::deque<bool> BoolContainer;

```

Comment:
while ```vector<bool>``` doesn't have everything as functional as a regular type vector,
have you actually found an issue using it?
Based on the code in this PR, it looks like  ```vector<bool>```  will work